sys/kern/subr_bus.c

```diff
@@ -5678,13 +5713,54 @@ static int
 devctl2_ioctl(struct cdev *cdev, u_long cmd, caddr_t data, int fflag,
     struct thread *td)
 {
+#if defined(COMPAT_FREEBSD32) || defined(COMPAT_FREEBSD64)
+	struct devreq dr;
+#endif
+#ifdef COMPAT_FREEBSD32
+	struct devreq32 *req32;
+#endif
+#ifdef COMPAT_FREEBSD64
+	struct devreq64 *req64;
+#endif
 	struct devreq *req;
 	device_t dev;
 	int error, old;
 
+#ifdef COMPAT_FREEBSD32
```

Comment:
FWIW, I don't think we really need freebsd32 support for these ioctls, but I guess it doesn't hurt.

Reply:
I agree it's not all that useful, but I've been making an habit of doing 32-bit versions where I can, loosely aiming in the direction of a 32-bit disk image working with only a kernel being replaced on a 64-bit kernel.